portfolio/src/main/webapp/script.js

```diff
@@ -12,119 +12,121 @@
-    }
-    image12.onclick = function(){
-        modal.style.display = "block";
-        modal_img.src = this.src;
-        description.innerHTML = this.alt;
-    }
-    image13.onclick = function(){
-        modal.style.display = "block";
-        modal_img.src = this.src;
-        description.innerHTML = this.alt;
-    }
-    image14.onclick = function(){
-        modal.style.display = "block";
-        modal_img.src = this.src;
-        description.innerHTML = this.alt;
-    }
-    image15.onclick = function(){
-        modal.style.display = "block";
-        modal_img.src = this.src;
-        description.innerHTML = this.alt;
-    }
+  var description = document.getElementById("caption");
 
-    var close = document.getElementsByClassName("close")[0];
-    close.onclick = function() {
-        modal.style.display= "none";
-    }
-}
+  image.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image2.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image3.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image4.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image5.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image6.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image7.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image8.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image9.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image10.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image11.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image12.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image13.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image14.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
+  image15.onclick = function(){
+    modal.style.display = "block";
+    modal_img.src = this.src;
+    description.innerHTML = this.alt;
+  }
```

Comment:
let imageArray = []
for (i = "1"; i <= "15"; i++) {
  var image = document.getElementById(i);
  image.onclick = function(){
    modal.style.display = "block";
    modal_img.src = this.src;
    description.innerHTML = this.alt;
  }
  imageAray.push(image);
}
Could we replace this with array?

Reply:
Yes it works, thank you! I was wondering how I could do it more efficiently just now!